client/packages/invoices/src/Returns/InboundDetailView/DetailView.tsx

```diff
@@ -9,25 +9,36 @@ import {
   useTranslation,
   createQueryParamsStore,
   DetailTabs,
+  useEditModal,
 } from '@openmsupply-client/common';
 import { ContentArea } from './ContentArea';
 import { Toolbar } from './Toolbar';
 import { AppBarButtons } from './AppBarButtons';
 import { InboundReturnLineFragment, useReturns } from '../api';
 import { AppRoute } from '@openmsupply-client/config';
 import { SidePanel } from './SidePanel/SidePanel';
-import { ActivityLogList } from 'packages/system/src';
+import { ActivityLogList } from '@openmsupply-client/system';
 import { Footer } from './Footer';
+import { InboundReturnItem } from '../../types';
+import { InboundReturnEditModal } from '../modals';
 
 export const InboundReturnDetailView: FC = () => {
   const { data, isLoading } = useReturns.document.inboundReturn();
   const t = useTranslation('distribution');
   const navigate = useNavigate();
 
-  // TODO: hook these up to modal
-  const onRowClick = () => {};
+  const {
+    onOpen,
+    onClose,
+    isOpen,
+    entity: itemId,
+    mode,
+  } = useEditModal<string>();
 
-  const onAddItem = () => {};
+  const onRowClick = (row: InboundReturnLineFragment | InboundReturnItem) =>
+    onOpen(row.itemId);
+
+  const onAddItem = () => onOpen();
```

Comment:
Why do we need this as a separate function definition? Can't you just have `onAddItem={onOpen}` below? 
Or are we going to be adding more to this function?